source/extensions/filters/network/rocketmq_proxy/codec.cc

```diff
@@ -0,0 +1,293 @@
+      }
+      cmd->custom_header_ = decodeResponseExtHeader(static_cast<ResponseCode>(code), header_struct,
+                                                    static_cast<RequestCode>(request_code));
+    } else {
+      cmd->custom_header_ = decodeExtHeader(static_cast<RequestCode>(code), header_struct);
+    }
+    return cmd;
+  } else {
+    ENVOY_LOG(warn, "Unsupported header serialization type");
+    has_error = true;
+    return nullptr;
+  }
+}
+
+CommandCustomHeaderPtr Decoder::decodeExtHeader(RequestCode code,
+                                                ProtobufWkt::Struct& header_struct) {
+  const auto& fieldValuePair = header_struct.fields();
+  switch (code) {
+  case RequestCode::SendMessage: {
+    ASSERT(fieldValuePair.contains("extFields"));
+    const auto& extFields = fieldValuePair.at("extFields");
+    std::unique_ptr<SendMessageRequestHeader> sendMsgExtHeader =
+        std::make_unique<SendMessageRequestHeader>();
+    sendMsgExtHeader->version_ = SendMessageRequestVersion::V1;
+    sendMsgExtHeader->decode(extFields);
+    return sendMsgExtHeader;
+  }
+  case RequestCode::SendMessageV2: {
+    ASSERT(fieldValuePair.contains("extFields"));
+    const auto& extFields = fieldValuePair.at("extFields");
+    std::unique_ptr<SendMessageRequestHeader> sendMsgExtHeader =
+        std::make_unique<SendMessageRequestHeader>();
+    sendMsgExtHeader->version_ = SendMessageRequestVersion::V2;
+    sendMsgExtHeader->decode(extFields);
+    return sendMsgExtHeader;
+  }
+
+  case RequestCode::GetRouteInfoByTopic: {
+    ASSERT(fieldValuePair.contains("extFields"));
+    const auto& extFields = fieldValuePair.at("extFields");
+    std::unique_ptr<GetRouteInfoRequestHeader> getRouteInfoRequestHeader =
+        std::make_unique<GetRouteInfoRequestHeader>();
+    getRouteInfoRequestHeader->decode(extFields);
+    return getRouteInfoRequestHeader;
+  }
+
+  case RequestCode::UnregisterClient: {
+    ASSERT(fieldValuePair.contains("extFields"));
+    const auto& extFields = fieldValuePair.at("extFields");
+    std::unique_ptr<UnregisterClientRequestHeader> unregisterClientRequestHeader =
+        std::make_unique<UnregisterClientRequestHeader>();
+    unregisterClientRequestHeader->decode(extFields);
+    return unregisterClientRequestHeader;
+  }
+
+  case RequestCode::GetConsumerListByGroup: {
+    ASSERT(fieldValuePair.contains("extFields"));
+    const auto& extFields = fieldValuePair.at("extFields");
+    std::unique_ptr<GetConsumerListByGroupRequestHeader> getConsumerListByGroupRequestHeader =
+        std::make_unique<GetConsumerListByGroupRequestHeader>();
+    getConsumerListByGroupRequestHeader->decode(extFields);
+    return getConsumerListByGroupRequestHeader;
+  }
+
+  case RequestCode::PopMessage: {
+    ASSERT(fieldValuePair.contains("extFields"));
+    const auto& extFields = fieldValuePair.at("extFields");
+    std::unique_ptr<PopMessageRequestHeader> popMessageRequestHeader =
```

Comment:
pop_message_request_header, please fix all local variable cases in this PR

Reply:
OK.